modules/javafx.graphics/src/main/java/com/sun/javafx/application/LauncherImpl.java

```diff
@@ -788,7 +780,7 @@ private static void launchApplication1(final Class<? extends Application> appCla
                 if (currentPreloader != null) {
                     if (simulateSlowProgress) {
                         for (int i = 0; i < 100; i++) {
-                            notifyProgress(currentPreloader, (double)i / 100.0);
+                            notifyProgress(currentPreloader, i / 100.0);
```

Comment:
Does Eclipse really flag this as a warning? It doesn't seem a particularly useful warning, since an explicit cast to double is a reasonable pattern that can improve clarity, even if it isn't needed. In this specific case it is pretty clear without the explicit cast that it will do what you want, but other places are not so clear.

Reply:
100.0 is a double, so the result is also a double

Reply:
Right. Which is why I said that in this specific case it (meaning the changed code) is clear without the explicit cases. So I don't object to this particular change. I just used it as the first example I found to point out an overall concern.

Reply:
It flags those as an "unnecessary cast", how it flags them is up to the settings (info, warning, error). The code will function the same with or without them, but every so often they do expose a problem where the author thought a cast did something, while in reality it did not.